draft-ietf-tls-esni.md

```diff
@@ -446,33 +491,120 @@ to harmonize these to make sure that we maintain key separation.]]
 
 This value is placed in an "encrypted_server_name" extension.
 
-The client MAY either omit the "server_name" extension or provide
-an innocuous dummy one (this is required for technical conformance
-with {{!RFC7540}}; Section 9.2.)
-
-If the server does not negotiate TLS 1.3 or above, then the client
-MUST abort the connection with an "unsupported_version" alert.
-If the server does not provide an "encrypted_server_name" extension
-in EncryptedExtensions, the client MUST abort the connection with
-an "illegal_parameter" alert. Moreover, it MUST check that
-ClientESNIInner.nonce matches the value of the
-"encrypted_server_name" extension provided by the server,
-and otherwise abort the connection with an "illegal_parameter"
-alert.
-
-## Client-Facing Server Behavior
+The client MUST place the value of ESNIKeys.public_name in the "server_name"
+extension. (This is required for technical conformance with {{!RFC7540}};
+Section 9.2.)
+
+### Handling the server response {#handle-server-response}
+
+If the server negotiates TLS 1.3 or above and provides an
+"encrypted_server_name" extension in EncryptedExtensions, the client
+then processes the extension's "response_type" field:
+
+- If the value is "esni_accept", the client MUST check that the extension's
+  "nonce" field matches ClientESNIInner.nonce and otherwise abort the
+  connection with an "illegal_parameter" alert. The client then proceeds
+  with the connection as usual, verifying the certificate against the desired
+  name.
+
+- If the value is "esni_retry_request", the client proceeds with the handshake,
+  verifying the certificate against ESNIKeys.public_name as described in
+  {{verify-public-name}}. If verification or the handshake fails, the client
+  MUST return a failure to calling application. It MUST NOT use the retry keys
```

Comment:
Nit: ...to _the_ calling application.

Reply:
Done. Also removed "as described below". That could be read as "as described below, here's how you avoid using the retry keys", which is weird.